components/Details/Details.tsx

```diff
@@ -27,15 +29,27 @@ export const Details = ({
     scope: currentScope,
     versions: { current, latest },
   } = useContext(DocsContext);
+  const router = useRouter();
   const scopes = useMemo(() => getScopes(scope), [scope]);
   const [isOpened, setIsOpened] = useState<boolean>(Boolean(opened));
   const isInCurrentScope = scopes.includes(currentScope);
+  const detailsId = title
+    ? title
+        .replace(/[&\/\\#,+()$~%.'":*?<>{}^;\d]/g, "")
+        .replace(/\s/g, "-")
+        .replace(/-$/, "")
+        .toLowerCase()
+    : "title";
+  const anchorInPath = getAnchor(router.asPath);
 
   useEffect(() => {
     if (scopes.length) {
       setIsOpened(isInCurrentScope);
     }
-  }, [scopes, isInCurrentScope]);
+    if (anchorInPath === detailsId) {
+      setIsOpened(true);
+    }
```

Comment:
This has the potential to set `isOpened` twice, and could be different values or the same value. 

Perhaps a different way to write this so it only sets it once:
```ts
    if (scopes.length && isInCurrentScope || anchorInPath === detailsId) {
      setIsOpened(true);
    } else {
      setIsOpened(false);
    }
```

Going further, do you want to run these checks when any of the 4 variables change? If not, it would be better to separate these concerns into separate `useEffects`. See https://reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns.